lib/opte/src/engine/parse.rs

```diff
@@ -205,6 +218,69 @@ pub struct GeneveOverV6<Q: ByteSlice> {
     pub inner_ulp: Ulp<Q>,
 }
 
+impl<Q: ByteSlice> ValidGeneveOverV6<Q> {
+    /// Return packet info about the inner frame.
+    #[inline]
+    pub fn ulp_meoi(
+        &self,
+        pkt_len: usize,
+    ) -> Result<mac_ether_offload_info_t, MeoiError> {
+        let adj_len = pkt_len
+            .checked_sub(
+                (
+                    &self.outer_eth,
+                    &self.outer_v6,
+                    &self.outer_udp,
+                    &self.outer_encap,
+                )
+                    .packet_length(),
+            )
+            .ok_or(MeoiError::PacketTooShort)?;
+        let meoi_len =
+            u32::try_from(adj_len).map_err(|_| MeoiError::PacketTooLong)?;
```

Comment:
Should this be an expect, rather than a handled error?

Reply:
Yeah, that could probably be justified here given that it's sourced from the underlying buffer lengths.

Reply:
Done in 81faf3d.